guide/src/migration.md

```diff
@@ -0,0 +1,183 @@
+# Appendix B: Migration Guides for major version changes
+
```

Comment:
It might be worth adding a link to the CHANGELOG.md here too.
```suggestion
This guide can help you upgrade code through breaking changes from one PyO3 version to the next. For a detailed list of all changes, see [CHANGELOG.md](https://github.com/PyO3/pyo3/blob/master/CHANGELOG.md)
```